### PR TITLE
FeatureFlag: support basic expressions (isDevMode)

### PR DESCRIPTION
**Pull request by ryantxu**
Original work from @oscarkilhed in:
https://github.com/grafana/grafana/compare/experimentals/feature-toggle-by-headers

**What this PR does / why we need it**:

We want some flags to be on by default in dev mode, this allows setting a default expression to use "isDevMode"

This PR uses a very big hammer (http://github.com/google/cel-go) to check if the system is in dev mode.  We could fix that with a simple string check, but this PR sets the stage for adding other static checks to the deployment (and eventually in context!)

TODO:
- [ ] Should we have all the build vars
- [ ] Should we add others?
- [ ] Should we punt on CEL for now?

### CODE REVIEW

File: pkg/services/featuremgmt/manager.go [hunk 1]
```diff
@@ -17,7 +20,7 @@ var (
 )
 
 type FeatureManager struct {
-	isDevMod  bool
+	isDevMode bool
```
Comment (ryantxu) [inline pkg/services/featuremgmt/manager.go:23]: 🤦 typo

**Comment by bergquist** [comment]
> We want some flags to be on by default in dev mode

Why? Wouldn't this just be annoying to debug for other developers?

**Comment by ryantxu** [comment]
> Why? Wouldn't this just be annoying to debug for other developers if it's not good enough.

@joshhunt   -- IIUC, it was for making loud validation errors.

I'll close for now, and if there is any interest, we can revisit

**Comment by joshhunt** [comment]
@bergquist @ryantxu Initial request was for https://github.com/grafana/grafana/pull/55732 - adding validation _warnings_ to save dashboard to get more developers "schema aware" and surface errors from changes sooner. "Being annoying" in a subtle and non-blocking way is kind of a goal for this

**Pull request closed without merging**